underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -148,50 +160,70 @@ public OutputStream create(String path) throws IOException {
   @Override
   public OutputStream create(String path, CreateOptions options) throws IOException {
     LOG.debug("Create method: {}", path);
-    String newPath = path.substring(Constants.HEADER_SWIFT.length());
-    if (newPath.endsWith("_SUCCESS")) {
+
+    // create will attempt to create the parent directory if it does not already exist
+    if (!mkdirs(getParentPath(path), true)) {
+      // fail if the parent directory does not exist and creation was unsuccessful
+      LOG.error("Parent directory creation unsuccessful for {}", path);
+      return null;
+    }
+
+    final String strippedPath = CommonUtils.stripPrefixIfPresent(path,
+        Constants.HEADER_SWIFT);
+    // TODO(adit): remove special handling of */_SUCCESS objects
+    if (strippedPath.endsWith("_SUCCESS")) {
       // when path/_SUCCESS is created, there is need to create path as
       // an empty object. This is required by Spark in case Spark
       // accesses path directly, bypassing Alluxio
-      String plainName = newPath.substring(0, newPath.indexOf("_SUCCESS"));
-      LOG.debug("Plain name: {}", plainName);
-      SwiftOutputStream out = SwiftDirectClient.put(mAccess, plainName);
-      out.close();
+      String plainName = CommonUtils.stripSuffixIfPresent(strippedPath, "_SUCCESS");
+      SwiftDirectClient.put(mAccess, plainName).close();
     }
-    return SwiftDirectClient.put(mAccess, newPath);
+    // We do not check if a folder with the same name exists
```

Comment:
This might not be a problem since it should be avoided by the Alluxio logic. We should later revisit what assumptions the ufs can make about operations and cut down on the required checks since they are expensive for many ufses.